### PR TITLE
added dwn and peergos to the comparison guide

### PR DESCRIPTION
**Pull request by andorsk**
This is intended to be a DRAFT. To discuss over next DIF DWN call. It's a starting point, but this will need to be reviewed, and then cleaned up, post alignment. 

After going through this exercise myself, I found it somewhat difficult sometimes to bucket into the categories that were defined. Not sure what the solution is here, but maybe we can change some of the categories a little. 

There's a few sections I think would be helpful to add in, and also need to revisit a few of the bullets. 

A few suggestions to consider and discuss over the next call:

1. Add sync as part of the evaluation criteria . 
2. Add supportable backends as part of the matrix.

### CODE REVIEW

**Comment by andorsk** [comment]
Shared over DIF call. To review between the next DIF call and provide feedback.

File: docs/companion_guide.md [hunk 1]
```diff
@@ -411,6 +411,31 @@ billion transactions per second (tps).
...
+| Query Capabilities | Peergos offers a RESTFul API with various capabilities described [here](https://book.peergos.org/features/apps.html). A few endpoints are directly specified. |
+
+
+| Decentralized Web Nodes	   | https://identity.foundation/decentralized-web-node/spec/ |
+| -------------- | ---------------------------------------------------------- |
+| Description	   | Decentralized Web Nodes are a mesh-like datastore construction that enable an entity to operate multiple nodes that sync to the same state across one another, enabling the owning entity to secure, manage, and transact their data with others without reliance on location or provider-specific infrastructure, interfaces, or routing mechanisms. |
+| Specification  |  [Open-Source Specification incubated by DIF](https://identity.foundation/decentralized-web-node/spec/) |
+| Deployment     | [ Pending ] |
+| Identity       | Decentralized Identifiers |
+| Authentication | Web3 Wallets and DID. |
```
Comment (LiranCohen) [inline docs/companion_guide.md:433]: I'm not sure if Web3 Wallets fit here, might be confusing with existing Web3 infrastructure.
Would this be a wallet/agent of it's own, a  DWN-Aware Wallet, a Web5 Wallet?

File: docs/companion_guide.md [hunk 2]
```diff
@@ -411,6 +411,31 @@ billion transactions per second (tps).
...
+| Transport      | [HTTP RESTful APIs](https://book.peergos.org/features/apps.html) |
+| Schema / Data Representation | [DAG CBOR Encoded IPLD Objects and Raw Objects](https://book.peergos.org/security/bats.html). JSON Schema for configuration. |
+| Query Capabilities | Peergos offers a RESTFul API with various capabilities described [here](https://book.peergos.org/features/apps.html). A few endpoints are directly specified. |
+
+
+| Decentralized Web Nodes	   | https://identity.foundation/decentralized-web-node/spec/ |
+| -------------- | ---------------------------------------------------------- |
+| Description	   | Decentralized Web Nodes are a mesh-like datastore construction that enable an entity to operate multiple nodes that sync to the same state across one another, enabling the owning entity to secure, manage, and transact their data with others without reliance on location or provider-specific infrastructure, interfaces, or routing mechanisms. |
+| Specification  |  [Open-Source Specification incubated by DIF](https://identity.foundation/decentralized-web-node/spec/) |
+| Deployment     | [ Pending ] |
```
Comment (LiranCohen) [inline docs/companion_guide.md:431]: I think we can probably add here that it will be offered as a multi-tenant SaaS service, as well as self-hosted.

**Review by LiranCohen — COMMENTED** [review]
Just some nits, but looks good.

File: docs/companion_guide.md (hunk 1 above)
Comment (andorsk) [inline docs/companion_guide.md:433]:
```suggestion
| Authentication | DWN Aware Wallets / DID based  |
```

File: docs/companion_guide.md (hunk 2 above)
Comment (andorsk) [inline docs/companion_guide.md:431]:
```suggestion
| Deployment     | Self Hosted or as a SaaS Multi-Tenant Service |
```

File: docs/companion_guide.md [hunk 3]
```diff
@@ -411,6 +411,31 @@ billion transactions per second (tps).
 | Schema / Data Representation | JSON Formatted |
 | Query Capabilities | [Not found in documentation] |
 
+| Peergos	   |  https://https://peergos.org/ |
+| -------------- | ---------------------------------------------------------- |
+| Description	   | Peergos is a decentralised protocol and open-source platform for storage, social media and applications |
+| Specification  | Proprietary specification |
```
Comment (ianopolous) [inline docs/companion_guide.md:417]: What is meant by proprietary here? Peergos protocol is 100% open.

File: docs/companion_guide.md [hunk 4]
```diff
@@ -411,6 +411,31 @@ billion transactions per second (tps).
...
+| Deployment     | Offered as a SaaS solution. Can be self hosted. |
+| Identity       | [PKI](https://book.peergos.org/security/pki.html) + [Centralized password management, with password based keypairs(password + username + public salt)](https://book.peergos.org/security/login.html) |
```
Comment (ianopolous) [inline docs/companion_guide.md:419]: identity keypairs are random. the password is use to retrieve your encrypted identity keypair from your DWN

File: docs/companion_guide.md [hunk 5]
```diff
@@ -411,6 +411,31 @@ billion transactions per second (tps).
 | Schema / Data Representation | JSON Formatted |
 | Query Capabilities | [Not found in documentation] |
 
+| Peergos	   |  https://https://peergos.org/ |
+| -------------- | ---------------------------------------------------------- |
+| Description	   | Peergos is a decentralised protocol and open-source platform for storage, social media and applications |
```
Comment (ianopolous) [inline docs/companion_guide.md:416]: You can also run multiple Peergos nodes that sync the same state across one another, enabling the owning entity to secure, manage, and transact their data with others without reliance on location or provider-specific infrastructure

File: docs/companion_guide.md (hunk 3 above)
Comment (andorsk) [reply docs/companion_guide.md:417]: @ianopolous happy to change it ( or would welcome a PR to update it if you'd be willing ). Could you please point to the specification so we can make the update?

File: docs/companion_guide.md (hunk 5 above)
Comment (andorsk) [reply docs/companion_guide.md:416]: @ianopolous there is no direct sync section in this guide, and we have considered putting it in, but haven't yet. If you think sync should be described in some section (i.e description ) please provide a PR with the update. The current description is taken directly off the website https://peergos.org/ @ https://peergos.org/about#:~:text=Peergos%20is%20a%20decentralised%20protocol,end%2Dto%2Dend%20encryption.

File: docs/companion_guide.md (hunk 4 above)
Comment (andorsk) [reply docs/companion_guide.md:419]: @ianopolous the Centralized password management, with password based keypairs(password + username + public salt) language was taken directly as language from https://book.peergos.org/security/login.html. If you think this wording should be updated, would welcome a PR describe what you think should be updated and why. We are happy to take a look and incorporate reasonable updates.

Comment (ianopolous) [reply docs/companion_guide.md:419]: Sounds like we need to make that clearer. :-) The "encrypted entry data" in that diagram, which contains the identity keypair and root capability to your filesystem) is stored only on your DWN. 
I can do a PR with these changes this week hopefully.

**Comment by andorsk** [comment]
Dear @ianopolous, we greatly appreciate your valuable feedback regarding Peergos. We are open to receiving suggestions or pull requests to address your concerns. While most of the active working group members have not directly interacted with Peergos, they acknowledge its significance as an ecosystem technology. Since you possess a deeper understanding of the technology, we warmly welcome your contributions and suggestions. Our aim is to accurately represent Peergos in the companion guide to the best of our ability. If you believe there are any misrepresentations or areas that require more clarity, please let us know, and we will be glad to incorporate any appropriate changes. PR's are ideal.

Thank you for taking the time to review the companion guide, and we look forward to receiving further feedback from you.

**Pull request merged**